src/portable/st/synopsys/dcd_synopsys.c

```diff
@@ -1241,34 +1236,41 @@ TU_ATTR_WEAK bool dcd_alloc_mem_for_conf(uint8_t rhport, tusb_desc_configuration
 
   // Determine number of used out EPs of current configuration and size of two biggest out EPs
   uint8_t nUsedOutEPs = 0, cnt_ep, cnt_tt;
-  bool tmp;
   uint16_t sz[2] = {0, 0};
 
   for (cnt_ep = 0; cnt_ep < EP_MAX; cnt_ep++)
   {
-    tmp = false;
     for (cnt_tt = 0; cnt_tt <= TUSB_XFER_INTERRUPT; cnt_tt++)
     {
-      tmp |= report.ep_transfer_type[cnt_ep][TUSB_DIR_OUT][cnt_tt];
+      if (report.ep_transfer_type[cnt_ep][TUSB_DIR_OUT][cnt_tt])
+      {
+        nUsedOutEPs++;
+        break;
+      }
     }
-    nUsedOutEPs += tmp;
 
     if (sz[0] < report.ep_size[cnt_ep][TUSB_DIR_OUT])
     {
       sz[1] = sz[0];
       sz[0] = report.ep_size[cnt_ep][TUSB_DIR_OUT];
     }
+    else if (sz[1] < report.ep_size[cnt_ep][TUSB_DIR_OUT])
+    {
+      sz[1] = report.ep_size[cnt_ep][TUSB_DIR_OUT];
+    }
   }
 
   // For configuration use the approach as explained in bus_reset()
-  _allocated_fifo_words = 15 + 2*nUsedOutEPs + (sz[0] / 4) + (sz[0] % 4 > 0 ? 1 : 0) + (sz[1] / 4) + (sz[1] % 4 > 0 ? 1 : 0) + 2;   // again, i do not really know why we need + 2 but otherwise it does not work
+  _allocated_fifo_words = 13 + 1 + 1 + 2 * nUsedOutEPs + ((sz[0] / 4) + (sz[1] / 4) + 1) + 2;   // again, i do not really know why we need + 2 but otherwise it does not work
```

Comment:
Even better might be `13 + 1 + 1 + 2 * nUsedOutEPs + (sz[0] + sz[1]) / 4 + (sz[0] + sz[1]) % 4 + 2`. Otherwise, if sz[0] = sz[1] = 3, we would allocate 1 word too less.

Reply:
I was using this recommendation:
```
If multiple isochronous endpoints are enabled, then at least two (Largest Packet Size / 4) + 1 spaces must be
allocated to receive back-to-back packets. Typically, two (Largest Packet Size / 4) + 1 spaces are recommended
so that when the previous packet is being transferred to the CPU, the USB can receive the subsequent packet.
```
I admit that I could interpret this in a way that ```+1``` is applied to both endpoint sizes and not to sum of them.
If packets are internally put at 32bits boundaries this could be put like this
```(sz[0] + 3) / 4) + (sz[1] + 3) / 4)```
... this would also cover case when sz contains 0 elements.

Reply:
or maybe we really need Largest (Packet Size / 4) + 1 (2 space for 4 bytes packet).
it would be better to change place where sz[] is filled with value and have
```sz[0] = (report.ep_size[cnt_ep][TUSB_DIR_OUT] / 4) + 1;```
then use simple sz[0] + sz[1] and that would fulfill recommendation for one or two max packets.

Reply:
Looks good! 👍